src/main/java/de/bonndan/nivio/output/icons/LocalIcons.java

```diff
@@ -3,33 +3,53 @@
 import de.bonndan.nivio.util.URLHelper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 import org.springframework.util.StringUtils;
 
+import java.io.File;
 import java.net.URL;
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
 
 import static de.bonndan.nivio.output.icons.IconMapping.DEFAULT_ICON;
 
+/**
+ * This component is responsible to resolve icons into urls / data urls.
+ */
 @Component
 public class LocalIcons {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(LocalIcons.class);
+    private static final String initErrorMsg = "Default icon could not be loaded from icon set folder %s." +
+            " Make sure all npm dependencies are installed (or run mvn package).";
+    private static final String DEFAULT_ICONS_FOLDER = "/static/icons/svg/";
 
     /**
      * default icon data url
      */
-    private String defaultIcon = null;
+    private final String defaultIcon;
+
+    private final String iconFolder;
 
     /**
      * data url cache
      */
     private final Map<String, String> iconDataUrls = new ConcurrentHashMap<>();
 
-    public LocalIcons() {
-        getIconUrl(DEFAULT_ICON.getIcon()).ifPresent(s -> defaultIcon = s);
+    /**
+     * @param iconFolder optional dir containing a different icon set
+     */
+    public LocalIcons(@Value("${nivio.iconFolder:#{null}}") String iconFolder) {
+        if (iconFolder != null) {
+            this.iconFolder = iconFolder.endsWith(File.separator) ? iconFolder : iconFolder + File.separator;
+        } else {
+            this.iconFolder = DEFAULT_ICONS_FOLDER;
+        }
+        defaultIcon = getIconUrl(DEFAULT_ICON.getIcon()).orElseThrow(() -> {
+            throw new RuntimeException(String.format(initErrorMsg, this.iconFolder));
+        });
```

Comment:
Okay, the problem I had was indeed resolved by obtaining the npm dependency. 👍

Reply:
Why not making the parameter `@Nonnull` and using the `DEFAULT_ICONS_FOLDER` as the fallback value?

Reply:
I've had to look up how that is done using injected values, but it works. Change is pushed.